src/resource.h

```diff
@@ -138,6 +139,18 @@ std::vector<typename T::Ptr> Resource::Package(Package::Ptr pkg) {
     return ts_pkgd;
   }
 
+  int approx_num_pkgs = quantity() / fill_mass;
+  // Check if the number of packages is within the limits, including if
+  // int overflow is reached
+  if (approx_num_pkgs > Package::SplitLimit() || approx_num_pkgs == std::numeric_limits<int>::min()) {
```

Comment:
I wonder about making a function `Package::ExceedsSplitLimit(approx_num_pkgs)` that can handle these two variations of ways to exceed the limit - and any other we find?